### PR TITLE
Add new csharp ms technology labels

### LINKED ISSUE #15454: Add new C# ms.technology values.
The new ms.technology labels for the C# guide have been approved.

These should be added to the docfx.json folder based metadata at the beginning of this sprint.

### PR DESCRIPTION
**Pull request by BillWagner**
Fixes #15454 

There are three commits to make it easier to review.
The first updated docfx.json with folder based metadata.
The second assigns file based metadata for those articles in the tutorials folder.
The third commit assigns file based metadata for those articles in the root C# guide folder.

Not every article is tagged perfectly, but the new values are correct, and as the reorganization progresses.

### CODE REVIEW

File: docs/csharp/basic-types.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 title: Basic Types - C# Guide
 description: Learn about the core types (numerics, strings, and object) in all C# programs
 ms.date: 10/10/2016
+ms.technology: csharp-fundamentals
```
Comment (mairaw) [inline docs/csharp/basic-types.md]: this one seems to be the most common under the csharp folder. add that to docfx json perhaps like the following and then this only has to be applied to the ones that are different?

```json
 "docs/csharp/**.md": "csharp-fundamentals",
```

File: docs/csharp/tutorials/working-with-linq.md [hunk 2]
```diff
@@ -2,6 +2,7 @@
 title: Working with LINQ
 description: This tutorial teaches you how to generate sequences with LINQ, write methods for use in LINQ queries, and distinguish between eager and lazy evaluation.
 ms.date: 10/29/2018
+ms.technology: csharp.linq
```
Comment (mairaw) [inline docs/csharp/tutorials/working-with-linq.md]: is this the value with . or should it be - ?

```suggestion
ms.technology: csharp-linq
```

**Review by mairaw — APPROVED** [review]
LGTM. Left a few suggestions.

File: docs/csharp/basic-types.md (hunk 1 above)
Comment (BillWagner) [reply docs/csharp/basic-types.md]: I like this thought, but I'm going to leave the tags in these files. As part of the reorg and new TOC, I'll be moving these to more appropriate locations. The `docs/csharp` folder won't have articles in it. This way, I have one less place to look as the new folder structure emerges.

Comment (mairaw) [reply docs/csharp/basic-types.md]: Makes sense!

**Pull request merged**